### PR TITLE
DBZ-5141 Should be sleep with REGISTRATION_RETRY_DELAY when registry MBean failed

### PR DESCRIPTION
**Pull request by harveyyue**
https://issues.redhat.com/browse/DBZ-5141

### CODE REVIEW

**Comment by jpechane** [comment]
@harveyyue Good catch! I had to be out of my mind yesterday :-(.

File: debezium-core/src/main/java/io/debezium/metrics/Metrics.java [hunk 1]
```diff
@@ -86,21 +89,23 @@ public synchronized void register() {
...
-                    if (attempt <= REGISTRATION_RETRIES) {
+                    if (attempt < REGISTRATION_RETRIES) {
                         LOGGER.warn(
                                 "Unable to register metrics as an old set with the same name exists, retrying in {} (attempt {} out of {})",
                                 REGISTRATION_RETRY_DELAY, attempt, REGISTRATION_RETRIES);
+                        metronome.pause();
                     }
                     else {
-                        LOGGER.error("Failed to register metrics MBean, metrics will not be available");
+                        LOGGER.error("Failed to register metrics MBean, metrics will not be available, retrying in {} (attempt {} out of {})",
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/metrics/Metrics.java:99]: PLease return back the original message. The retry should not happen after {{else}} branch.

File: debezium-core/src/main/java/io/debezium/metrics/Metrics.java [hunk 2]
```diff
@@ -72,6 +74,7 @@ protected Metrics(CommonConnectorConfig connectorConfig, String contextName, boo
      */
     public synchronized void register() {
         try {
+            final Metronome metronome = Metronome.sleeper(REGISTRATION_RETRY_DELAY, Clock.system());
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/metrics/Metrics.java:77]: Could you please resturcture the code to mimic https://github.com/debezium/debezium/blob/main/debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java#L345 ?

File: debezium-core/src/main/java/io/debezium/metrics/Metrics.java (hunk 1 above)
Comment (harveyyue) [reply debezium-core/src/main/java/io/debezium/metrics/Metrics.java:99]: @jpechane, welcome. Got it.

File: debezium-core/src/main/java/io/debezium/metrics/Metrics.java (hunk 2 above)
Comment (harveyyue) [reply debezium-core/src/main/java/io/debezium/metrics/Metrics.java]: Done.

**Comment by jpechane** [comment]
@harveyyue Could you please return back the `<` you used in the condition? I belive you were correct on that one change.

**Comment by harveyyue** [comment]
> @harveyyue Could you please return back the `<` you used in the condition? I belive you were correct on that one change.

Had returned to `<`

**Pull request merged**